sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.webpubsub.client;
+
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.util.BinaryData;
+import com.azure.messaging.webpubsub.client.implementation.AckMessage;
+import com.azure.messaging.webpubsub.client.implementation.MessageDecoder;
+import com.azure.messaging.webpubsub.client.implementation.MessageEncoder;
+import com.azure.messaging.webpubsub.client.implementation.JoinGroupMessage;
+import com.azure.messaging.webpubsub.client.implementation.LeaveGroupMessage;
+import com.azure.messaging.webpubsub.client.implementation.SendToGroupMessage;
+import jakarta.websocket.ClientEndpointConfig;
+import jakarta.websocket.CloseReason;
+import jakarta.websocket.Endpoint;
+import jakarta.websocket.EndpointConfig;
+import jakarta.websocket.MessageHandler;
+import jakarta.websocket.Session;
+import org.glassfish.tyrus.client.ClientManager;
+import org.glassfish.tyrus.core.CloseReasons;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks;
+import reactor.core.scheduler.Schedulers;
+
+import java.net.URI;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicLong;
+
+@ServiceClient(builder = WebPubSubClientBuilder.class)
+public class WebPubSubAsyncClient {
+
+    private final Mono<String> clientAccessUriProvider;
+
+    private final ClientManager clientManager;
+
+    private Endpoint endpoint;
+
+    private Session session;
+
+    private Sinks.Many<WebPubSubMessage> messageSink = Sinks.many().multicast().onBackpressureBuffer();
+
+    WebPubSubAsyncClient(Mono<String> clientAccessUriProvider) {
+        this.clientAccessUriProvider = clientAccessUriProvider;
+
+        this.clientManager = ClientManager.createClient();
+    }
+
+    public Mono<Void> start() {
+        this.endpoint = new ClientEndpoint();
+        ClientEndpointConfig config = ClientEndpointConfig.Builder.create()
+            .preferredSubprotocols(Collections.singletonList("json.webpubsub.azure.v1"))
+            .encoders(Collections.singletonList(MessageEncoder.class))
+            .decoders(Collections.singletonList(MessageDecoder.class))
+            .build();
+
+        return clientAccessUriProvider.flatMap(uri -> Mono.fromCallable(() -> {
+            this.session = clientManager.connectToServer(endpoint, config, new URI(uri));
+            return (Void) null;
+        }).subscribeOn(Schedulers.boundedElastic()));
+    }
+
+    public Mono<Void> stop() {
+        return Mono.fromCallable(() -> {
+            if (session != null && session.isOpen()) {
+                session.close(CloseReasons.NORMAL_CLOSURE.getCloseReason());
+                messageSink.tryEmitComplete();
+                messageSink = Sinks.many().multicast().onBackpressureBuffer();
```

Comment:
Should we send complete to sink?

And is it correct to create a new sink after complete?

Reply:
I think it depends on how we want users to use it. If `stop()` emit the complete signal, it should complete the `receive()` downstream subscriber, and the user may need to re-subscribe the `receive()` when starting again. 
If we don't complete in stop(), user may don't need to re-subscribe and continue to receive after we call `start()`.

Reply:
TBD is the ack stream. We probably want it sync to connectionId. I.e., if the connection can be recovered, ack keep the stream; but if connection is stopped or it cannot be recovered after disconnect, complete be sent.